pkg/utils/utils.go

```diff
@@ -258,3 +258,46 @@ func CleanCachedNetConf(cRefPath string) error {
 	}
 	return nil
 }
+
+func SaveAllocatedPCI(pciAddress, ns, dataDir string) error {
+	allocatedPCIPath := filepath.Join(dataDir, "pci")
+	if err := os.MkdirAll(allocatedPCIPath, 0600); err != nil {
+		return fmt.Errorf("failed to create the sriov data directory(%q): %v", dataDir, err)
+	}
+
+	path := filepath.Join(allocatedPCIPath, pciAddress)
+
+	err := os.WriteFile(path, []byte(ns), 0600)
+	if err != nil {
+		return fmt.Errorf("failed to write used PCI address lock file in the path(%q): %v", path, err)
+	}
+
+	return err
+}
+
+func CleanAllocatedPCI(pciAddress, dataDir string) error {
+	path := filepath.Join(dataDir, "pci", pciAddress)
+	if err := os.Remove(path); err != nil {
+		return fmt.Errorf("error removing PCI address lock file %s: %q", path, err)
+	}
+	return nil
+}
+
+func IsAllocated(pciAddress, dataDir string) (bool, string, error) {
+	path := filepath.Join(dataDir, "pci", pciAddress)
+	_, err := os.Stat(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return false, "", nil
+		}
+
+		return false, "", fmt.Errorf("failed to check for pci address file for %s: %v", pciAddress, err)
+	}
+
+	dat, err := os.ReadFile(path)
+	if err != nil {
+		return false, "", fmt.Errorf("failed to read for pci address file for %s: %v", pciAddress, err)
```

Comment:
Please, add the whole path to the error. It makes it easier to debug in case of errors.

```suggestion
		return false, "", fmt.Errorf("failed to read PCI address file %s: %w", path, err)
```